framework/src/org/apache/cordova/CordovaWebViewImpl.java

```diff
@@ -176,22 +176,24 @@ public void run() {
                     e.printStackTrace();
                 }
 
-                // If timeout, then stop loading and handle error
-                if (loadUrlTimeout == currentLoadUrlTimeout) {
+                // If timeout, then stop loading and handle error (if activity still exists)
+                if (loadUrlTimeout == currentLoadUrlTimeout && cordova.getActivity() != null) {
```

Comment:
I wonder if we should add an else condition to log a warning/debug statement when the activity no longer exists. This may help provide better insights for developers when they hit these edge cases unexpectedly.

e.g:

```java
if (loadUrlTimeout == currentLoadUrlTimeout && cordova.getActivity() != null) {
   ...
} else if (cordova.getActivity() == null) {
    LOG.d(TAG, "Cordova activity does not exists.");
}

Reply:
Ah sure, indeed. I'll add those logs!